base/applications/rapps/settings.cpp

```diff
@@ -0,0 +1,247 @@
+    infoFields.AddTail(new SettingsFieldString((settings.szNoProxyFor), MAX_PATH, L"NoProxyFor"));
+    infoFields.AddTail(new SettingsFieldBool(&(settings.bUseSource), L"bUseSource"));
+    infoFields.AddTail(new SettingsFieldString((settings.szSourceURL), INTERNET_MAX_URL_LENGTH, L"SourceURL"));
+
+    return;
+}
+
+BOOL SaveAllSettings(CRegKey &key, SETTINGS_INFO &settings)
+{
+    BOOL bAllSuccess = TRUE;
+    ATL::CAtlList<SettingsField *> infoFields;
+
+    AddInfoFields(infoFields, settings);
+
+    POSITION InfoListPosition = infoFields.GetHeadPosition();
+    while (InfoListPosition)
+    {
+        SettingsField *Info = infoFields.GetNext(InfoListPosition);
+        if (!Info->Save(key))
+        {
+            bAllSuccess = FALSE;
+            // TODO: error log
+        }
+        delete Info;
+    }
+    return bAllSuccess;
+}
+
+BOOL LoadAllSettings(CRegKey &key, SETTINGS_INFO &settings)
+{
+    BOOL bAllSuccess = TRUE;
+    ATL::CAtlList<SettingsField *> infoFields;
+
+    AddInfoFields(infoFields, settings);
+
+    POSITION InfoListPosition = infoFields.GetHeadPosition();
+    while (InfoListPosition)
+    {
+        SettingsField *Info = infoFields.GetNext(InfoListPosition);
+        if (!Info->Load(key))
+        {
+            bAllSuccess = FALSE;
+            // TODO: error log
+        }
+        delete Info;
+    }
+    return bAllSuccess;
+}
+
+VOID FillDefaultSettings(PSETTINGS_INFO pSettingsInfo)
+{
+    ATL::CStringW szDownloadDir;
+    ZeroMemory(pSettingsInfo, sizeof(SETTINGS_INFO));
+
+    pSettingsInfo->bSaveWndPos = TRUE;
+    pSettingsInfo->bUpdateAtStart = FALSE;
+    pSettingsInfo->bLogEnabled = TRUE;
+    pSettingsInfo->bUseSource = FALSE;
+
+    if (FAILED(SHGetFolderPathW(NULL, CSIDL_PERSONAL, NULL, SHGFP_TYPE_CURRENT, szDownloadDir.GetBuffer(MAX_PATH))))
+    {
+        szDownloadDir.ReleaseBuffer();
+        if (!szDownloadDir.GetEnvironmentVariableW(L"SystemDrive"))
+        {
+            szDownloadDir = L"C:";
+        }
+    }
+    else
+    {
+        szDownloadDir.ReleaseBuffer();
+    }
+
+    PathAppendW(szDownloadDir.GetBuffer(MAX_PATH), L"\\RAPPS Downloads");
+    szDownloadDir.ReleaseBuffer();
+
+    ATL::CStringW::CopyChars(pSettingsInfo->szDownloadDir,
+        _countof(pSettingsInfo->szDownloadDir),
+        szDownloadDir.GetString(),
+        szDownloadDir.GetLength() + 1);
+
+    pSettingsInfo->bDelInstaller = FALSE;
+    pSettingsInfo->Maximized = FALSE;
+    pSettingsInfo->Left = CW_USEDEFAULT;
+    pSettingsInfo->Top = CW_USEDEFAULT;
+    pSettingsInfo->Width = 680;
+    pSettingsInfo->Height = 450;
+}
+
+BOOL LoadSettings(PSETTINGS_INFO pSettingsInfo)
+{
+    ATL::CRegKey RegKey;
+    if (RegKey.Open(HKEY_CURRENT_USER, L"Software\\ReactOS\\rapps", KEY_READ) != ERROR_SUCCESS)
+    {
+        return FALSE;
+    }
+
+    BOOL bRet = LoadAllSettings(RegKey, *pSettingsInfo);
+    RegKey.Close();
```

Comment:
This is automatically done by the class, hence why using classes like this is more convenient :)